shared/src/commonMain/kotlin/io/projectnewm/login/LoginFieldValidator.kt

```diff
@@ -0,0 +1,28 @@
+package io.projectnewm.login
+
+class LoginFieldValidator {
+    public companion object {
+        public fun validate(email: String, password: String): Boolean {
+            return isValidEmail(email) && isValidPassword(password)
+        }
+
+        private fun isValidEmail(email: CharSequence?): Boolean {
+            val emailAddressRegex = Regex(
+                "[a-zA-Z0-9\\+\\.\\_\\%\\-\\+]{1,256}" +
+                        "\\@" +
+                        "[a-zA-Z0-9][a-zA-Z0-9\\-]{0,64}" +
+                        "(" +
+                        "\\." +
+                        "[a-zA-Z0-9][a-zA-Z0-9\\-]{0,25}" +
+                        ")+"
+            )
+            return email?.matches(emailAddressRegex) ?: false
+        }
+
+        private fun isValidPassword(password: String?): Boolean {
+            val password = password?.let { it } ?: return false
```

Comment:
It's cleanest to just do 
```suggestion
           return if (password == null || password.isBlank()) false else true
```